wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -164,22 +164,31 @@ def store_path(
         )
         start_time = None
         multi = False
-        try:
-            objs[0].load()
-            # S3 doesn't have real folders, however there are cases where the folder key has a valid file which will not
-            # trigger a recursive upload.
-            # we should check the object's metadata says it is a directory and do a multi file upload if it is
-            if "x-directory" in objs[0].content_type:
-                multi = True
-        except self._botocore.exceptions.ClientError as e:
-            if e.response["Error"]["Code"] == "404":
-                multi = True
-            else:
-                raise CommError(
-                    "Unable to connect to S3 ({}): {}".format(
-                        e.response["Error"]["Code"], e.response["Error"]["Message"]
+        if key == "":
+            start_time = time.time()
+            termlog(
+                'Generating checksum for up to %i objects in the bucket "%s"... '
+                % (max_objects, bucket),
+                newline=False,
+            )
+            objs = self._s3.Bucket(bucket).objects.limit(max_objects)
```

Comment:
We need to pass `multi=True` on line 203 in this case as well, but you can't set `multi=True` here because then you'd hit the branch on line 192. This needs to be reorganized a bit.